requests_kerberos/kerberos_.py

```diff
@@ -118,9 +121,16 @@ def generate_request_header(self, response, host, is_preemptive=False):
             # w/ name-based HTTP hosting)
             kerb_host = self.hostname_override if self.hostname_override is not None else host
             kerb_spn = "{0}@{1}".format(self.service, kerb_host)
+            
+            kwargs = {'principal': self.principal}
+            # kerberos-sspi: Never pass principal. Raise if user tries to specify one.
+            if self._using_kerberos_sspi:
+                kwargs = {}
+                if self.principal:
+                    raise NotImplementedError("Can't use 'principal' argument with kerberos-sspi.")
```

Comment:
Ok, the logic here is getting a bit torturous. Maybe initialize `kwargs` to empty dict, then do if/else for the two cases (adding principal to the dict, raising `NotImplementedError`).